coalib/misc/Exceptions.py

```diff
@@ -20,10 +15,10 @@ def get_exitcode(exception, log_printer=None):
         exitcode = exception.code
     elif isinstance(exception, VersionConflict):
         log_message = Constants.VERSION_CONFLICT_MESSAGE % str(exception.req)
-        log_printer.log_exception(log_message, exception)
+        LogPrinterMixin().log_exception(log_message, exception)
```

Comment:
Makes sense. If there are more helper functions needed, then we might need to shift to class. Its more maintainable and readable.
Anyways, work for this PR seems to be almost done(apart from bugs, if any in this commit).
Its a long commit. Please review it as you get time.